src/sympc/module/nn/functional.py

```diff
@@ -37,3 +49,327 @@ def mse_loss(pred: MPCTensor, target: MPCTensor, reduction: str = "mean") -> MPC
+
+    if isinstance(stride, int):
+        stride = (stride, stride)
+
+    if len(stride) != 2:
+        raise ValueError("Stride should have only 2 dimensions")
+
+    if isinstance(padding, int):
+        padding = (padding, padding)
+
+    if padding[0] > kernel_size[0] or padding[1] > kernel_size[1]:
+        raise ValueError("Padding should be <= kernel_size / 2")
+
+    if len(padding) != 2:
+        raise ValueError("Padding should have only 2 dimensions")
+
+    if isinstance(dilation, int):
+        dilation = (dilation, dilation)
+
+    if len(dilation) != 2:
+        raise ValueError("Dilation should have only 2 dimensions")
+
+    if dilation[0] != 1 or dilation[1] != 1:
+        raise ValueError("Supported only dilation == 1")
+
+    return kernel_size, stride, padding, dilation
+
+
+def _reshape_max_pool2d(
+    x: MPCTensor,
+    kernel_size: Tuple[int, int],
+    stride: Tuple[int, int],
+    padding: Tuple[int, int],
+    dilation: Tuple[int, int],
+) -> MPCTensor:
+    """Prepare the share tensors by calling the reshape function in parallel at each party.
+
+    Args:
+        x (MPCTensor): the MPCTensor on which to apply the reshape operation
+        kernel_size (Tuple[int, int]): the kernel size
+        stride (Tuple[int, int]): the stride size
+        padding (Tuple[int, int]): the padding size
+        dilation (Tuple[int, int]): the dilation size
+
+    Returns:
+        The reshaped MPCTensor.
+    """
+    session = x.session
+
+    args = [[share, kernel_size, stride, padding, dilation] for share in x.share_ptrs]
+    shares = parallel_execution(helper_max_pool2d_reshape, session.parties)(args)
+
+    res_shape = shares[0].shape.get()
+    res = MPCTensor(shares=shares, session=session, shape=res_shape)
+    return res
+
+
+def helper_max_pool2d_reshape(
+    x: ShareTensor,
+    kernel_size: Tuple[int, int],
+    stride: Tuple[int, int],
+    padding: Tuple[int, int],
+    dilation: Tuple[int, int],
+) -> ShareTensor:
+    """Function that runs at each party for preparing the share.
+
+    Reshape each share tensor to prepare it for calling 'argmax'.
+    The new share would have "each element" as the input on which we
+    will run the max_pool2d kernel.
+
+    Args:
+        x (ShareTensor): the ShareTensor on which to apply the reshaping
+        kernel_size (Tuple[int, int]): the kernel size
+        stride (Tuple[int, int]): the stride size
+        padding (Tuple[int, int]): the padding size
+        dilation (Tuple[int, int]): the dilation size
+
+    Returns:
+        The prepared share tensor (reshaped)
+    """
+    session = get_session(x.session_uuid)
+    tensor = x.tensor.numpy()
+
+    padding = [(0, 0)] * len(tensor.shape[:-2]) + [
+        (padding[0], padding[0]),
+        (padding[1], padding[1]),
+    ]
+    tensor_type = session.tensor_type
+
+    padding_value = 0
+    if session.rank == 0:
+        # ATTENTION: Min value for max_pool2d that works -25
+        padding_value = -25
```

Comment:
Curious, what does this do?

Reply:
Select a small value for the padding.
Or could you be more specific?